frontend/document/document.lua

```diff
@@ -414,9 +414,18 @@ function Document:getDrawnImagesStatistics()
 end
 
 function Document:getPagePart(pageno, rect, rotation)
-    local tile = self:renderPage(pageno, rect, 1, rotation, 1, 0)
-    local target = Blitbuffer.new(rect.w, rect.h, self.render_color and self.color_bb_type or nil)
-    target:blitFrom(tile.bb, 0, 0, rect.x, rect.y, rect.w, rect.h)
+    local canvas_size = CanvasContext:getSize()
+    local zoom = math.min(canvas_size.w / rect.w, canvas_size.h / rect.h)
```

Comment:
The zoom factor can also make the resulting picture smaller. I imagine we don't want to enforce a minimum zoom factor of 1 (that could be huge! sometimes), so a stupidly simple workaround to both that and @poire-z's comment https://github.com/koreader/koreader/pull/6709#issuecomment-698624533 could also be something like:
```suggestion
    local zoom = math.min(canvas_size.w*2 / rect.w, canvas_size.h*2 / rect.h)
```

Reply:
Or of course:
```suggestion
    local zoom = math.max(canvas_size.w / rect.w, canvas_size.h / rect.h)
```